### PR TITLE
feat: expand integration test coverage for edge cases

### LINKED ISSUE #15: Expand integration test coverage
The integration test suite (via `ShellTest` harness in `tests/`) needs broader coverage before new features land.

**Add integration tests for:**
- [ ] `cd` to nonexistent directories (expect `ShellError::FileNotFound`)
- [ ] `cd` to a file (expect `ShellError::NotADirectory`)
- [ ] `exit` with non-zero codes
- [ ] `type` for PATH executables (once I/O is wired through `ShellIo`)
- [ ] Multiple sequential commands in a single session
- [ ] Empty/whitespace-only input lines

### PR DESCRIPTION
**Pull request by cdprice02**
## Description

Closes #15.

Adds `tests/edge_cases.rs` with 8 integration tests using the `ShellTest` harness covering:
- `cd` to a nonexistent directory — asserts stderr contains "no such file" or "not found"
- `cd` to a file (not a directory) — asserts stderr contains "not a directory"
- `exit 42` — asserts `exit_code() == 42`
- `type echo` — asserts output contains "echo" and "builtin"
- `type sh` (Unix) / `type cmd` (Windows) — asserts output contains `<name> is` and a path separator
- Multiple sequential `echo` commands — asserts both outputs appear
- Whitespace-only input lines — asserts no errors and shell continues
- Empty script — asserts no errors

## Type of change
- [x] New feature (test coverage)

## Test checklist
- [x] `cargo test` passes
- [x] `cargo clippy --all-targets --all-features -- -D warnings` passes
- [x] New tests added in `tests/edge_cases.rs`

### CODE REVIEW

File: tests/edge_cases.rs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        .with_isolated_home()
+        .script("cd /nonexistent_ferrish_test_path_xyz")
+        .run();
+
+    let error = result.error().to_lowercase();
+    assert!(
+        error.contains("no such file") || error.contains("not found"),
+        "Expected 'no such file' or 'not found' in stderr, got: {}",
+        result.error()
+    );
```
[the remark this answers is not in the document]
Comment (cdprice02) [reply tests/edge_cases.rs]: Fixed in ba7e1db: removed the broad 'not found' fallback. We went further and dropped this test entirely since it duplicates `test_cd_to_nonexistent_directory_shows_error` in builtin.rs, which already uses the precise `assert_error_contains("no such file or directory")` assertion.

File: tests/edge_cases.rs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+// ============================================================================
+
+/// exit with code 42 propagates exactly 42 as the process exit code
+#[test]
+fn test_exit_nonzero_code_42() {
+    let result = ShellTest::new().script("exit 42").run();
+    assert_eq!(result.exit_code(), 42, "exit 42 should produce exit code 42");
+}
+
+// ============================================================================
```
[the remark this answers is not in the document]
Comment (cdprice02) [reply tests/edge_cases.rs]: Fixed in ba7e1db: removed all tests that duplicate existing coverage in builtin.rs and repl.rs. edge_cases.rs now contains only two genuinely novel scenarios: `test_type_builtin_echo_shows_builtin` (tests that `echo` is identified as a builtin, distinct from the existing `type exit` test) and `test_whitespace_only_lines_no_error` (spaces+tabs mix, not covered elsewhere).
